server/src/calendar/continous_scraping.rs

```diff
@@ -0,0 +1,287 @@
+                from: lower_middle + chrono::Days::new(1),
+                to: self.to,
+            },
+        )
+    }
+}
+
+#[cfg(test)]
+mod test_scrape_order {
+    use super::ScrapeOrder;
+    use chrono::NaiveDate;
+    #[test]
+    fn test_split() {
+        let order = ScrapeOrder::new(("".to_string(), 0), 2020, 1);
+        let (o1, o2) = order.split();
+        assert_eq!(order.from, NaiveDate::from_ymd_opt(2020, 1, 1).unwrap());
+        assert_eq!(order.to, NaiveDate::from_ymd_opt(2020, 12, 31).unwrap());
+        assert_eq!(o1.from, order.from);
+        assert_eq!(o2.to, order.to);
+        assert_eq!(o1.to + chrono::Duration::days(1), o2.from);
+    }
+    #[test]
+    fn test_split_small() {
+        let order = ScrapeOrder {
+            key: "".to_string(),
+            room_id: 0,
+            from: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+            to: NaiveDate::from_ymd_opt(2020, 1, 2).unwrap(),
+        };
+        let (o1, o2) = order.split();
+        assert_eq!(o1.to + chrono::Duration::days(1), o2.from);
+        assert_eq!(order.num_days(), 2);
+        assert_eq!(order.from, o1.from);
+        assert_eq!(order.to, o2.to);
+        assert_eq!(o1.num_days(), 1);
+        assert_eq!(o2.num_days(), 1);
+        assert_eq!(order.from, o1.to);
+        assert_eq!(order.to, o2.from);
+    }
+    #[test]
+    fn test_num_days() {
+        let mut order = ScrapeOrder {
+            key: "".to_string(),
+            room_id: 0,
+            from: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+            to: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+        };
+        assert_eq!(order.num_days(), 1);
+        order.to = NaiveDate::from_ymd_opt(2020, 1, 2).unwrap();
+        assert_eq!(order.num_days(), 2);
+        order.to = NaiveDate::from_ymd_opt(2020, 12, 31).unwrap();
+        assert_eq!(order.num_days(), 366);
+    }
+    #[test]
+    fn test_same_day() {
+        let order = ScrapeOrder::new(("".to_string(), 0), 2020, 0);
+        assert_eq!(order.from, NaiveDate::from_ymd_opt(2020, 1, 1).unwrap());
+        assert_eq!(order.to, NaiveDate::from_ymd_opt(2019, 12, 31).unwrap());
+        assert_eq!(order.num_days(), 0);
+    }
+}
+
+async fn scrape(
+    client: &Client,
+    id: (String, i32),
+    from_year: i32,
+    year_duration: i32,
+) -> Result<usize, usize> {
+    // request and parse the xml file
+    let mut request_queue = vec![ScrapeOrder::new(id, from_year, year_duration)];
+    let mut success_cnt = 0;
+    let mut retry_smaller_was_nessesary = false;
+    while !request_queue.is_empty() {
+        let mut new_request_queue = vec![];
+        for work in request_queue {
+            let events = XMLEvents::request(client, work.clone()).await;
+
+            //store the events in the database if successful, otherwise retry
+            match events {
+                Ok(events) => {
+                    success_cnt += events.len();
+                    events.store();
```

Comment:
I think `store_in_db()` would be a bit better to understand

Reply:
agree